docs/src/contribute.md

```diff
@@ -0,0 +1,85 @@
+
+### [Incorporating the feature requests through a fork](@id con-feat_req-fork)
+
+!!! note
+    The approach used for providing code is based on the excellent description of the [JuMP](https://jump.dev/JuMP.jl/stable/developers/contributing/#Contribute-code-to-JuMP) package.
+    We essentially follow the same approach with minor changes.
+
+If you would like to work directly in `TimeStruct`, you can also incorporate your changes directly.
+In this case, it is beneficial to follow the outlined steps:
+
+#### [Step 1: Create an issue](@id con-feat_req-fork-step_1)
+
+Even if you plan to incorporate the code directly, we advise you to first follow the steps outlined in *[Create an Issue](@ref con-feat_req-issue)*.
+This way, it is possible for us to comment on the solution approach(es) and assess potential problems with other time structures.
+
+Through creating an issue first, it is possible for us to comment directly on the proposed changes and assess, whether we consider the proposed changes to follow the philosophy of the framework.
+
+#### [Step 2: Create a fork of `TimeStruct`](@id con-feat_req-fork-step_2)
+
+Contributing code to `TimeStruct` should follow the standard approach by creating a fork of the repository.
+All work on the code should occur within the fork.
+
+#### [Step 3: Checkout a new branch in your local fork](@id con-feat_req-fork-step_3)
+
+It is in general preferable to work on a separate branch when developing new components.
+
+#### [Step 4: Make changes to the code base](@id con-feat_req-fork-step_4)
+
+Incorporate your changes in your new branch.
+The changes should be commented to understand the thought process behind them.
+In addition, please provide new tests for the added functionality and be certain that the existing tests run.
+New tests should be based on a minimum working example in which the new concept is evaluated.
+
+!!! tip
+    It is in our experience easiest to use the package [`TestEnv`](https://github.com/JuliaTesting/TestEnv.jl) for testing the complete package.
+
+Aside from the individual tests, it is required to use [`JuliaFormatter`](https://domluna.github.io/JuliaFormatter.jl/stable/) on the code.
+
+It is not necessary to provide changes directly in the documentation.
+It can be easier to include these changes after the pull request is accepted in principal.
+
+#### [Step 5: Create a pull request](@id con-feat_req-fork-step_5)
+
+Once you are satisified with your changes, create a pull request towards the main branch of the `TimeStruct` repository.
+We will internally assign the relevant person to the pull request.
+
+You may receive quite a few comments with respect to the incorporation and how it may potentially affect other parts of the code.
+Please remaing patient as it may take potentially some time before we can respond to the changes, although we try to answer as fast as possible.
```

Comment:
```suggestion
Please remain patient as it may take potentially some time before we can respond to the changes, although we try to answer as fast as possible.
```